File: templates/platform/android/DeviceInfoChannel.kt
package {{package}}.channels

import android.os.Build
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel

object DeviceInfoChannel {
    private const val NAME = "{{channel_prefix}}/device"

    fun register(messenger: BinaryMessenger) {
        MethodChannel(messenger, NAME).setMethodCallHandler { call, result ->
            when (call.method) {
                "getDeviceInfo" -> result.success(
                    mapOf(
                        "platform" to "Android",
                        "version" to Build.VERSION.RELEASE,
                        "model" to Build.MODEL,
                        "sdk" to Build.VERSION.SDK_INT
                    )
                )
                else -> result.notImplemented()
            }
        }
    }
}

File: templates/platform/ios/DeviceInfoChannel.swift
import Flutter

struct DeviceInfoChannel {
  static let name = "{{channel_prefix}}/device"

  static func register(with messenger: FlutterBinaryMessenger) {
    let channel = FlutterMethodChannel(name: name, binaryMessenger: messenger)
    channel.setMethodCallHandler { call, result in
      switch call.method {
      case "getDeviceInfo":
        let device = UIDevice.current
        result([
          "platform": "iOS",
          "version": device.systemVersion,
          "model": device.model,
          "name": device.name
        ])
      default:
        result(FlutterMethodNotImplemented)
      }
    }
  }
}
